include/nbl/builtin/hlsl/colorspace/EOTF.hlsl

```diff
@@ -0,0 +1,114 @@
+
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_COLOR_SPACE_EOTF_INCLUDED_
+
+#include <nbl/builtin/hlsl/common.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace colorspace
+{
+namespace eotf
+{
+
+float3 identity(in float3 nonlinear)
+{
+    return nonlinear;
+}
+
+float3 impl_shared_2_4(in float3 nonlinear, in float vertex)
+{
+    bool3 right = greaterThan(nonlinear, vertex.xxx);
```

Comment:
I'm not a big fan of this .xxx thing, if you were to just put `vertex` here for example, it would splat it into a float3. What do you think @devshgraphicsprogramming

Reply:
I use `.xxx` so it can be identified as a promoted scalar